src/services/puzzle/puzzle-router.ts

```diff
@@ -0,0 +1,209 @@
+        return res.status(StatusCode.SuccessOK).send(updatedPuzzleItem);
+    } catch (error) {
+        return next(new RouterError(StatusCode.ServerErrorInternal, "InternalError"));
+    }
+});
+
+/**
+ * @api {post} /puzzle/create POST /puzzle/create
+ * @apiGroup Puzzle
+ * @apiDescription Create a puzzle item for user.
+ *
+ * @apiHeader {String} Authorization User's JWT Token with admin permissions.
+ *
+ * @apiBody {json} teamName The user's requested team name.
+ * @apiParamExample {Json} Request Body Example for a teamName:
+ * {
+ *      "teamName": "myTeam"
+ * }
+ *
+ * @apiSuccess (200: Success) {Json} Success!.
+ *
+ * @apiError (404: User Not Found) {String} UserNotFound User does not exist.
+ * */
+puzzleRouter.post("/create", strongJwtVerification, async (req: Request, res: Response, _: NextFunction) => {
+    const payload: JwtPayload = res.locals.payload as JwtPayload;
+    const targetUser: string = payload.id;
+    const teamName: string = req.body.teamName;
+
+    // no need to check that user exists, handled by strongJwtVerification
+
+    try {
+        // Create a new PuzzleItem model
+        const newPuzzleItem = new Models.PuzzleItem({
+            userId: targetUser,
+            teamName: teamName,
+            lastCorrect: -1,
+            problemComplete: [false, false, false, false, false, false, false, false, false, false],
+        });
+
+        await Models.PuzzleItem.findOneAndUpdate({ userId: targetUser }, newPuzzleItem, { upsert: true, new: true });
+
+        return res.status(StatusCode.SuccessOK).send({ newPuzzleItem });
+    } catch (error) {
+        return res.status(StatusCode.ServerErrorInternal).send({ status: false, error: "Internal Server Error" });
+    }
+});
+
+/**
+ * @api {get} /puzzle/ranking GET /puzzle/ranking
+ * @apiGroup Puzzle
+ * @apiDescription Get the current ranking of participants in puzzle.
+ *
+ * @apiSuccess (200: Success) {Array} ranking List of all teamnames ranked.
+ * @apiSuccess (200: Success) {String} qrInfo Array of stringified QR codes for the requested item 
+
+ * @apiSuccessExample Example Success Response:
+ * 	HTTP/1.1 200 OK
+ *	[
+ *      "team1",
+ *      "team2",
+ *      "team3",
+ *      "team4",
+ *      "team5"
+ *  ]
+ *
+ * @apiUse weakVerifyErrors
+ */
+puzzleRouter.get("/ranking", weakJwtVerification, async (_1: Request, res: Response, _2: NextFunction) => {
+    const teamsRanked: PuzzleItem[] = await Models.PuzzleItem.aggregate([
+        {
+            $addFields: {
+                trueCount: {
+                    $sum: {
+                        $map: {
+                            input: "$problemComplete",
+                            as: "complete",
+                            in: { $cond: ["$complete", Config.TRUE_VALUE, Config.FALSE_VALUE] },
+                        },
+                    },
+                },
+            },
+        },
+        { $sort: { trueCount: -1, lastCorrect: 1 } },
+    ]);
```

Comment:
Is there really no better way of doing this? We might be better off just pulling all data and doing a projection, then sorting by that on API side? 

We can also use sort on the cursor that gets returned once we perform a find query, will lead to cleaner code